animation_data.cpp

```diff
@@ -20,25 +25,76 @@ std::vector<SDL_Rect> BuildAnimationRects(SDL_Rect start,
 
   return result;
 }
+
+// Takes the number of frames and sets the offset for each frame of the
+// animation to zero.
+std::vector<Location> NoOffsets(int number_of_frames) {
+  std::vector<Location> result;
+  for (int i = 0; i < number_of_frames; i++) {
+    result.push_back({0, 0});
+  }
+
+  return result;
+}
 }  // namespace
 
 AnimationMap GetGokuAnimations() {
   return {{AnimationType::kDefault,
            {.type = AnimationType::kDefault,
             .time_between_frames_ms = 300,
+            .offsets = NoOffsets(/*number_of_frames*/ 2),
             .image_rects =
                 BuildAnimationRects(kGokuIdleStart, /*number_of_frames=*/2),
             .priority = 0}},
           {AnimationType::kRunRight,
            {.type = AnimationType::kRunRight,
             .time_between_frames_ms = 100,
+            .offsets = NoOffsets(/*number_of_frames*/ 2),
             .image_rects =
                 BuildAnimationRects(kGokuRunRightStart, /*number_of_frames=*/2),
             .priority = 1}},
           {AnimationType::kRunLeft,
            {.type = AnimationType::kRunLeft,
             .time_between_frames_ms = 100,
+            .offsets = NoOffsets(/*number_of_frames=*/2),
             .image_rects =
                 BuildAnimationRects(kGokuRunLeftStart, /*number_of_frames=*/2),
-            .priority = 1}}};
+            .priority = 1}},
+          {AnimationType::kSSIdle,
+           {.type = AnimationType::kSSIdle,
+            .time_between_frames_ms = 300,
+            .offsets = NoOffsets(/*number_of_frames=*/2),
+            .image_rects =
+                BuildAnimationRects(kSSGokuIdleStart, /*number_of_frames=*/2),
+            .priority = 0}},
+          {AnimationType::kSSRunRight,
+           {.type = AnimationType::kSSRunRight,
+            .time_between_frames_ms = 100,
+            .offsets = NoOffsets(/*number_of_frames=*/2),
+            .image_rects = BuildAnimationRects(kSSGokuRunRightStart,
+                                               /*number_of_frames=*/2),
+            .priority = 1}},
+          {AnimationType::kSSRunLeft,
+           {.type = AnimationType::kSSRunLeft,
+            .time_between_frames_ms = 100,
+            .offsets = NoOffsets(/*number_of_frames=*/2),
+            .image_rects = BuildAnimationRects(kSSGokuRunLeftStart,
+                                               /*number_of_frames=*/2),
+            .priority = 1}},
+
+          {AnimationType::kSSTransformation,
+           {.type = AnimationType::kSSTransformation,
+            .time_between_frames_ms = 100,
+            .offsets = {{-29, -55},
+                        {-30, -55},
+                        {-29, -55},
+                        {-29, -55},
+                        {-29, -55},
+                        {-29, -55},
+                        {-29, -55},
+                        {-29, -55},
+                        {-29, -55}},
+            .image_rects = BuildAnimationRects(kGokuSSTransformationStart,
```

Comment:
make a version of BuildAnimationRects that takes the offsets as an argument instead of the number of frames:

For illustration I'm calling the struct with the rect and offsets an "AnimationFrame":

```
std::vector<AnimationFrame> BuildAnimationRects(SDL_Rect start, std::vector<Location> offsets) { ... }
```

Use this version when there are offsets, and the other one will be for when all the offsets are 0, 0.